certora/specs/Delegation.spec

```diff
@@ -0,0 +1,58 @@
+import "ERC20.spec";
+
+ghost mathint sumOfVotingPower {
+    init_state axiom sumOfVotingPower == 0;
+}
+
+// Slot for DelegationTokenStorage._delegatedVotingPower
+hook Sstore (slot 0x669be2f4ee1b0b5f3858e4135f31064efe8fa923b09bf21bf538f64f2c3e1101)[KEY address addr] uint256 newValue (uint256 oldValue) {
+    sumOfVotingPower = sumOfVotingPower - oldValue + newValue;
+}
+
+// Check that zero address has no voting power assuming that zero address can't make transactions.
+invariant zeroAddressNoVotingPower()
+    delegatee(0x0) == 0x0 && delegatedVotingPower(0x0) == 0
+    { preserved with (env e) { require e.msg.sender != 0; } }
+
+// Check that the voting power plus the virtual voting power of address zero is equal to the total supply of tokens.
+invariant totalSupplyIsSumOfVirtualVotingPower()
+    to_mathint(totalSupply()) == sumOfVotingPower + currentContract._zeroVirtualVotingPower
+    {
+      preserved {
+          // Safe requires because the proxy contract should be initialized right after construction.
+          require totalSupply() == 0;
+          require sumOfVotingPower == 0;
```

Comment:
I don't understand the comment nor why it's ok to require this

Reply:
We expect the contract to be used behind a proxy, and the proxy to call `initialize` when it is constructed. The comment explains that, and highlights the fact that no calls to the contract are made in between construction and initialization

Reply:
I'll @QGarchery change this comment, if need be, as he's the original author.

Reply:
Is it clearer now @MathisGD ? If it is, what would you change in the comment ?

Reply:
issue with the fact that it's required for every functino and not only for initialized

Reply:
Fixed in https://github.com/morpho-org/morpho-token-upgradeable/pull/92/commits/afe61a8ca1f210dc70e5c00a3c3383d2b1462baf